Browser/keywords/interaction.py

```diff
@@ -138,48 +142,45 @@ def type_secret(
         ``clear`` Set to false, if the field shall not be cleared before typing.
         Defaults to true.
 
-        Notice that if you use the keyword like:
-        | Type Secret    id=password_field    password
-        the password is shown as a normal keyword argument. A way to avoid that is
-        using variables like
-        | Type Secret    id=password_field    ${PASSWORD}
-
-        Please notice that Robot Framework logs all arguments using the TRACE
-        level and tests must not be executed using level below DEBUG if the
-        password should not be logged in any format.
-
         See `Type Text` for details.
         """
+        secret = self._resolve_secret(environment_variable)
         self._type_text(selector, secret, delay, clear, log_response=False)
 
     @keyword(tags=["Setter", "PageContent"])
-    def fill_secret(self, selector: str, secret: str):
-        """Fills the given ``secret`` into the text field found by ``selector``.
+    def fill_secret(self, selector: str, environment_variable: str):
+        """Fills the given secret from ``environment_variable`` into the
+        text field found by ``selector``.
 
-        The difference to `Fill Text` is that this keyword does not log secret
-        on the INFO level. If ``enable_playwright_debug`` is enabled in the library
+        This keyword does not log secret in Robot Framework logs.

```

Comment:
This is not correct from users point of view. Although the keyword does not log it, RF will log it, if log level is high enough. But for normal users it is not possible to see the difference between the library keyword and RF core.